vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion_transfer.cc

```diff
@@ -3,115 +3,70 @@
-  for (auto& promotion : promotions) {
-    if (!promotion) {
-      continue;
-    }
-
-    ids.push_back(promotion->id);
-  }
-
-  ledger_->database()->GetSpendableUnblindedTokensByTriggerIds(
-      ids,
-      callback);
+void PromotionTransfer::Start(
+    ledger::PostSuggestionsClaimCallback callback) const {
+  ledger_->database()->GetSpendableUnblindedTokens(
+      std::bind(&PromotionTransfer::OnGetSpendableUnblindedTokens, this, _1,
+                std::move(callback)));
 }
 
-void PromotionTransfer::Start(ledger::PostSuggestionsClaimCallback callback) {
-  auto tokens_callback =
-      std::bind(&PromotionTransfer::OnGetEligibleTokens, this, _1, callback);
-
-  GetEligibleTokens(tokens_callback);
-}
-
-void PromotionTransfer::OnGetEligibleTokens(
-    type::UnblindedTokenList list,
-    ledger::PostSuggestionsClaimCallback callback) {
+void PromotionTransfer::OnGetSpendableUnblindedTokens(
+    type::UnblindedTokenList tokens,
+    ledger::PostSuggestionsClaimCallback callback) const {
   std::vector<type::UnblindedToken> token_list;
-  for (auto& item : list) {
-    token_list.push_back(*item);
+  for (auto& token : tokens) {
+    token_list.push_back(*token);
   }
 
   if (token_list.empty()) {
-    callback(type::Result::LEDGER_OK, "");
-    return;
+    return callback(type::Result::LEDGER_OK, "");
   }
 
   credential::CredentialsRedeem redeem;
   redeem.type = type::RewardsType::TRANSFER;
   redeem.processor = type::ContributionProcessor::BRAVE_TOKENS;
-  redeem.token_list = token_list;
+  redeem.token_list = std::move(token_list);
 
-  const double transfer_amount = token_list.size() * constant::kVotePrice;
   credentials_->DrainTokens(
-      redeem, [this, transfer_amount, callback](const type::Result result,
-                                                std::string drain_id) {
-        if (result == type::Result::LEDGER_OK) {
-          ledger_->database()->SaveEventLog(log::kPromotionsClaimed,
-                                            std::to_string(transfer_amount));
-        }
-        callback(result, drain_id);
-      });
+      redeem, std::bind(&PromotionTransfer::OnDrainTokens, this, _1, _2,
+                        redeem.token_list.size() * constant::kVotePrice,
+                        std::move(callback)));
 }
 
-std::vector<type::PromotionType> PromotionTransfer::GetEligiblePromotions() {
-  std::vector<type::PromotionType> promotions = {
-    type::PromotionType::ADS
-  };
-
-  const bool claim_ugp =
-      ledger_->ledger_client()->GetBooleanOption(option::kClaimUGP);
-  if (claim_ugp) {
-    promotions.push_back(type::PromotionType::UGP);
+void PromotionTransfer::OnDrainTokens(
+    type::Result result,
+    std::string drain_id,
+    double transfer_amount,
+    ledger::PostSuggestionsClaimCallback callback) const {
+  if (result == type::Result::LEDGER_OK) {
+    ledger_->database()->SaveEventLog(log::kPromotionVBATDrained,
+                                      std::to_string(transfer_amount));
```

Comment:
We should use `base::NumberToString` from `base/strings/string_number_conversions.h`.